types/index.d.ts

```diff
@@ -2697,7 +2699,86 @@ declare namespace math {
     to(x: Unit | MathArray | Matrix, unit: Unit | string): Unit | MathArray | Matrix;
 
     /*************************************************************************
-     * Utils functions
+     * Utils
+     ************************************************************************/
+     isNumber(x: unknown): x is number;
+
+     isBigNumber(x: unknown): x is BigNumber;
+
+     isComplex(x: unknown): x is Complex;
+
+     isFraction(x: unknown): x is Fraction;
+
+     isUnit(x: unknown): x is Unit;
+
+     isString(x: unknown): x is string;
+
+     isArray: ArrayConstructor['isArray'];
+
+     isMatrix(x: unknown): x is Matrix;
+
+     isCollection(x: unknown): x is (Matrix | any[]);
+
+     isDenseMatrix(x: unknown): x is Matrix;
+
+     isSparseMatrix(x: unknown): x is Matrix;
+
+     isRange(x: unknown): boolean;
+
```

Comment:
Currently, `math.Range` is exported, and returns a Range object. (I think that's the only way to get one.) In the spirit of matching up the TypeScript typing with the actual JavaScript semantics, the Range type should be declared in this file and this should be changed to a guard for Range. The only question is whether that should be done in this PR, or if it is out of scope for this PR and a separate issue should be filed. Thoughts?

Reply:
Fine with me to address this in a separate PR (to prevent too much scope creep in this PR 😉). But feel free to address right away if you're in the mood for it @ChristopherChudzicki